service/create/service.go

```diff
@@ -222,8 +227,14 @@ func (s *Service) runMachine(machine node.Node, clusterName, name string) error
 	// TODO(nhlfr): Check whether the instance has correct parameters. That will be most probably done when we
 	// will introduce the interface for creating, deleting and updating resources.
 	if instances.Reservations != nil {
-		s.logger.Log("info", fmt.Sprintf("instance '%s' already exists", name))
-		return nil
+		for _, r := range instances.Reservations {
+			for _, i := range r.Instances {
+				if *i.State.Code != EC2TerminatedState {
+					s.logger.Log("info", fmt.Sprintf("instance '%s' already exists", name))
+					return nil
+				}
+			}
+		}
```

Comment:
I just leave a comment here. This piece of code is only interested in structures the process holds in memory. There is no external dependency involved. So you have an algorithm that calculates some result based on some quite important knowledge. It would be good to have this logic in a separate function which can easily be tested. The test to filter instances is then small and nice. The code can be reused if necessary.